x-pack/plugins/alerting/server/routes/create_rule.ts

```diff
@@ -40,15 +39,26 @@ export const bodySchema = schema.object({
     interval: schema.string({ validate: validateDurationSchema }),
   }),
   actions: actionsSchema,
-  notify_when: schema.maybe(schema.nullable(schema.string({ validate: validateNotifyWhenType }))),
+  notify_when: schema.maybe(
+    schema.nullable(
+      schema.oneOf(
+        [
+          schema.literal('onActionGroupChange'),
+          schema.literal('onActiveAlert'),
+          schema.literal('onThrottleInterval'),
+        ],
+        { validate: validateNotifyWhenType }
+      )
+    )
+  ),
```

Comment:
This was causing the weird conversion below: `notify_when: rule.notify_when as RuleNotifyWhenType`